In [13]:
# ignore warning
import warnings
warnings.filterwarnings("ignore")

import numpy as np 
import pandas as pd 
import matplotlib.pyplot as plt
import seaborn as sns
data = pd.read_csv("Admission SAT.csv")

In [14]:
data=data["SAT"]
data

0      1363
1      1792
2      1954
3      1653
4      1593
       ... 
163    1722
164    1750
165    1555
166    1524
167    1461
Name: SAT, Length: 168, dtype: int64

In [11]:
# Set the number of bootstrap samples to generate
n_samples = 1000

# Create an empty array to store the bootstrap samples
bootstrap_samples = np.empty((n_samples, len(data)))
len(bootstrap_samples[0])

168

In [16]:
# Set the number of bootstrap samples to generate
n_samples = 1000

# Create an empty array to store the bootstrap samples
bootstrap_samples = np.empty((n_samples, len(data)))
len(bootstrap_samples[0])
# Generate the bootstrap samples
for i in range(n_samples):
    # Draw a sample of the same size as the original dataset, with replacement
    bootstrap_sample = np.random.choice(data, size=len(data), replace=True)

#here we are taking out random samples of size 168(data length) and it takes out a number place it in list replace it and continues



    # Store the bootstrap sample in the array
    bootstrap_samples[i] = bootstrap_sample

# Calculate the mean of each bootstrap sample
means = np.mean(bootstrap_samples, axis=1)


# Calculate the 95% confidence interval for the mean
ci_low = np.percentile(means, 2.5)
ci_high = np.percentile(means, 97.5)


In [17]:
means

array([1703.31547619, 1685.32142857, 1699.46428571, 1670.25      ,
       1697.27380952, 1673.04761905, 1700.38095238, 1687.9702381 ,
       1712.0297619 , 1703.88095238, 1689.375     , 1697.26785714,
       1669.14880952, 1701.97619048, 1697.85119048, 1684.32142857,
       1686.35714286, 1693.94047619, 1673.32142857, 1671.375     ,
       1693.39880952, 1715.9047619 , 1694.89880952, 1696.17261905,
       1699.        , 1681.14880952, 1700.89285714, 1712.25      ,
       1685.98214286, 1710.33928571, 1704.71428571, 1700.56547619,
       1688.49404762, 1650.45238095, 1702.72619048, 1700.0297619 ,
       1673.08928571, 1704.10119048, 1680.25595238, 1689.38690476,
       1683.66071429, 1695.07142857, 1711.2202381 , 1674.70238095,
       1706.57142857, 1716.375     , 1686.73214286, 1690.67261905,
       1698.20833333, 1708.47619048, 1704.41666667, 1713.74404762,
       1687.42261905, 1714.16666667, 1700.45833333, 1703.89880952,
       1716.42857143, 1697.47619048, 1691.94642857, 1691.16666

In [18]:
# Calculate the standard error of the mean
se_mean = np.std(means)

se_mean

14.236664918974679

In [8]:
ci_low,ci_high

(1667.584375, 1722.5254464285715)

This means that we are 95% confident that the true population mean score falls within the interval from 1667.584375  to 1722.525446

### EXPLAINATION

Resampling and bootstrap are both techniques used in statistics to analyze data and estimate population parameters.

Resampling is a general term used to describe the process of repeatedly drawing samples from a dataset to obtain additional information about the population. Resampling techniques can be broadly divided into two categories: parametric and non-parametric resampling. Parametric resampling assumes a particular probability distribution for the data, while non-parametric resampling methods do not make any assumptions about the underlying distribution.

Bootstrap is a type of non-parametric resampling technique that involves randomly sampling from the original dataset with replacement to create multiple "bootstrap samples." Each bootstrap sample is the same size as the original dataset, and may contain repeated observations. The goal of the bootstrap is to estimate the variability of a statistic of interest (such as the mean or median) by calculating that statistic on each bootstrap sample.

Therefore, bootstrap is a specific example of non-parametric resampling. While other non-parametric resampling techniques may involve different methods for selecting samples, bootstrap always involves random sampling with replacement